src/wasm-interpreter.h

```diff
@@ -2908,6 +2923,9 @@ class ModuleRunnerBase : public ExpressionRunner<SubType> {
     virtual void trap(const char* why) = 0;
     virtual void hostLimit(const char* why) = 0;
     virtual void throwException(const WasmException& exn) = 0;
+    // Get the Tag instance for a tag implemented in the host, that is, not
+    // among the linked ModuleRunner instances.
+    virtual Tag* getHostTag(Name name) { WASM_UNREACHABLE("missing host tag"); }
```

Comment:
Shouldn't this be a general `getImportedTag`? Host tags shouldn't be special for any reason and it would be best to be consistent in how we handle imports. Specifically, we should follow the same patterns for `getImportedFunction` and whatever we do for tags.

Reply:
(And btw we should also follow the same shared pattern for tables and memories as well.)

Reply:
We have these special host tags in the fuzzer, which we don't for functions - so this pattern is very convenient here, but maybe not for functions.

Perhaps this could be refactored to make these host tags come from a special internal module, I considered that but it was a lot more work, and I wasn't sure it would lead to benefits.

I do agree it makes sense to unify this, but I'm not sure which way, so I'd like to leave that for later.

Reply:
Ok, I tried to do that in the last two commits, PTAL

1. I thought you meant to generate an internal module etc. - that would be a lot more work, obviously.
2. But I don't quite think I follow you, as the loop in `getCanonicalTag` is still necessary?

Reply:
If `getCanonicalTag` calls out to the ExternalInterface to retrieve imported tags, then in multi-instance scenarios that ExternalInterface will look up the instance providing the import and call `getExportedTag` on it (which might in turn call `externalInterface->getImportedTag` again if it is a re-exported tag). So the loop is replaced with a recursion.